glusterapi/volume.py

```diff
@@ -1,132 +1,263 @@
-from common import BaseAPI
-from exceptions import GlusterApiInvalidInputs
+"""This module contains the python  glusterd2 volume api's implementation."""
+import httplib
+import json
 
+from glusterapi.common import BaseAPI, validate_uuid, validate_volume_name
+from glusterapi.exceptions import GlusterApiInvalidInputs
 
-class TransportType:
+
+class TransportType(object):
     TCP = "tcp"
     RDMA = "rdma"
     Both = "tcp,rdma"
 
+    def check(self, transport_type):
+        if transport_type not in (self.TCP, self.RDMA, self.Both):
+            raise GlusterApiInvalidInputs(
+                "Transport type %s not supported" % transport_type)
+
+
+def validate_brick(bricks=None):
+    """
+    Validate brick pattern.
+
+    :param bricks: (string) in the form of nodeid:brickpath
+    :return brick_req: (list) list of bricks
+    """
+    brick_req = []
+    if bricks is None:
+        return None
+    for brick in bricks:
```

Comment:
Madhu, the param bricks you document as type string and you iterate it here. Did you mean to say type as (list)?

Reply:
Thanks for catching this, missed updating doc